app/uk/gov/hmrc/ctcguaranteebalancerouter/services/AccessCodeService.scala

```diff
@@ -51,25 +46,20 @@ class AccessCodeServiceImpl @Inject() (connectorProvider: EISConnectorProvider)
     ec: ExecutionContext
   ): EitherT[Future, AccessCodeError, Unit] =
     for {
-      result          <- connectorProvider(countryCode).postAccessCodeRequest(grn, hc).leftMap(handleRoutingError)
-      validAccessCode <- extractAccessCode(result)
-      _               <- validateCode(accessCode, validAccessCode)
+      result <- connectorProvider(countryCode).postAccessCodeRequest(grn, hc).leftMap(handleRoutingError)
+      _      <- validateCode(accessCode, result.accessCode)
     } yield ()
 
-  private def extractAccessCode(value: JsValue)(implicit ec: ExecutionContext): EitherT[Future, AccessCodeError, AccessCode] =
-    value.validate[AccessCodeResponse] match {
-      case JsSuccess(value, _) => EitherT.rightT(value.accessCode)
-      case JsError(_)          => EitherT.leftT(AccessCodeError.InvalidJson)
-    }
-
   private def validateCode(providedCode: AccessCode, expectedCode: AccessCode)(implicit ec: ExecutionContext): EitherT[Future, AccessCodeError, Unit] =
     if (providedCode == expectedCode) EitherT.rightT(())
     else EitherT.leftT(AccessCodeError.InvalidAccessCode)
 
   // EIS will only return select status codes, so we need to infer from there
-  private def handleRoutingError(error: RoutingError): AccessCodeError = error match {
-    case RoutingError.Upstream(UpstreamErrorResponse(_, statusCode, _, _)) if statusCode <= 499 => AccessCodeError.NotFound
-    case x                                                                                      => AccessCodeError.Routing(x)
+  private def handleRoutingError(error: ConnectorError): AccessCodeError = error match {
+    case ConnectorError.NotFound                   => AccessCodeError.NotFound
+    case ConnectorError.FailedToDeserialise        => AccessCodeError.FailedToDeserialise
```

Comment:
I think we'll need our ConvertError and PresentationError classes but that can be implemented when the controller is implemented.

Reply:
Yup, that's when they'll happen.